es-app/src/SystemData.cpp

```diff
@@ -63,6 +63,23 @@ void SystemData::setIsGameSystemStatus()
 	mIsGameSystem = (mName != "retropie");
 }
 
+// test to see if a file is hidden in *nix (dot-prefixed)
+// could be expanded to check for Windows hidden attribute
+bool isHidden(const fs::path &filePath)
+{
+	if(Settings::getInstance()->getBool("ShowHiddenFiles"))
+		return false;
```

Comment:
It doesn't seem logical to me that the check would be in the function. The function should always do the operation - the check should be done on the code that calls the